packages/oidc-client/src/main.ts

```diff
@@ -0,0 +1,83 @@
+import * as core from '@actions/core'
+import * as actions_http_client from '@actions/http-client'
+import {
+  createHttpClient,
+  isSuccessStatusCode,
+  getApiVersion
+} from './internal/utils'
+import jwt_decode from 'jwt-decode'
+import {getIDTokenFromEnv, getIDTokenUrl} from './internal/config-variables'
+
+export async function getIDToken(audience: string): Promise<string> {
+  try {
+    //Check if id token is stored in environment variable
+
+    let id_token: string = getIDTokenFromEnv()
+    if (id_token !== undefined && id_token !== '') {
+      const secondsSinceEpoch = Math.round(Date.now() / 1000)
+      const id_token_json: any = jwt_decode(id_token)
+      if ('exp' in id_token_json) {
+        if (id_token_json['exp'] - secondsSinceEpoch > 300) {
+          // Expiry time is more than 5 mins
```

Comment:
by default the ID_TOKEN lifetime will be 5 min, so this will never be true.


And we have corner case here, since aud is input. we should not check only time. 
if aud is different it's a new id token the client needs.

Reply:
Is there any telemetry we can capture here? does actions framework have any standard capture of telemetry?